File: backend/src/main/java/com/example/backend/Model/Class/CarModel.java
package com.example.backend.Model.Class;

import jakarta.persistence.*;
import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Cacheable
@Data
@Entity
@Table(name = "CAR_MODEL")
public class CarModel {
    @Id
    private String id;
    @Column(name = "name")
    private String name;
    @Column(name = "generation")
    private Integer generation;
    @Column(name = "release_year")
    private Integer release_year;
}

File: frontend/src/app/_model/_interface/car-model.ts
export interface CarModel {
  id: string;
  name: string;
  generation: number;
  release_year: number;
}
export function isCarModel(obj: any): obj is CarModel {
  return (
    !!obj &&
    typeof obj === 'object' &&
    'release_year' in obj &&
    'generation' in obj &&
    typeof obj.release_year === 'number' &&
    typeof obj.generation === 'number' &&
    'id' in obj &&
    'name' in obj
  );
}
